lib/fetch/file.js

```diff
@@ -282,10 +279,7 @@ function processBlobParts (parts, options) {
 
       // 3. Append the result of UTF-8 encoding s to bytes.
       bytes.push(encoder.encode(s))
-    } else if (
-      types.isAnyArrayBuffer(element) ||
-      types.isTypedArray(element)
-    ) {
+    } else if (ArrayBuffer.isView(element) || types.isArrayBuffer(element)) {
```

Comment:
@KhafraDev Do you allow DataView here?

Reply:
The specification appears to allow it.

Reply:
lets use the existing BufferSource webidl converter

Reply:
Is this change correct?
I wanted to ask this because types.isTypedArray does not detect DataView.

Reply:
- typedef ([BufferSource](https://webidl.spec.whatwg.org/#BufferSource) or [Blob](https://w3c.github.io/FileAPI/#dfn-Blob) or [USVString](https://webidl.spec.whatwg.org/#idl-USVString)) BlobPart; ->
- typedef ([ArrayBufferView](https://webidl.spec.whatwg.org/#ArrayBufferView) or [ArrayBuffer](https://webidl.spec.whatwg.org/#idl-ArrayBuffer)) [BufferSource](https://webidl.spec.whatwg.org/#BufferSource); ->
- typedef ([Int8Array](https://webidl.spec.whatwg.org/#idl-Int8Array) or [Int16Array](https://webidl.spec.whatwg.org/#idl-Int16Array) or [Int32Array](https://webidl.spec.whatwg.org/#idl-Int32Array) or
         [Uint8Array](https://webidl.spec.whatwg.org/#idl-Uint8Array) or [Uint16Array](https://webidl.spec.whatwg.org/#idl-Uint16Array) or [Uint32Array](https://webidl.spec.whatwg.org/#idl-Uint32Array) or [Uint8ClampedArray](https://webidl.spec.whatwg.org/#idl-Uint8ClampedArray) or
         [BigInt64Array](https://webidl.spec.whatwg.org/#idl-BigInt64Array) or [BigUint64Array](https://webidl.spec.whatwg.org/#idl-BigUint64Array) or
         [Float32Array](https://webidl.spec.whatwg.org/#idl-Float32Array) or [Float64Array](https://webidl.spec.whatwg.org/#idl-Float64Array) or [DataView](https://webidl.spec.whatwg.org/#idl-DataView)) [ArrayBufferView](https://webidl.spec.whatwg.org/#ArrayBufferView);

Yes, a dataview is allowed. We should use `webidl.converters.BufferSource` instead of reimplementing it here.

Reply:
Thank you.
Then this code does not need to be changed.

Reply:
Yeah, I might add `webidl.is.BufferSource(value)` because we do the same checks multiple times.